test/integration/api/api_test.go

```diff
@@ -0,0 +1,43 @@
+package api_test
+
+import (
+	"bytes"
+	"net/http"
+	"soarca/internal/controller"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func initializeSoarca(t *testing.T) {
+	err := controller.Initialize()
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func TestCorsHeader(t *testing.T) {
```

Comment:
test also for restricted domains, and test if fails with non allowed domains

Reply:
done!